numpy/testing/_private/utils.py

```diff
@@ -688,8 +688,10 @@ def assert_array_compare(comparison, x, y, err_msg='', verbose=True,
     __tracebackhide__ = True  # Hide traceback for py.test
     from numpy.core import array, array2string, isnan, inf, bool_, errstate, all, max, object_
 
-    x = array(x, copy=False, subok=True)
-    y = array(y, copy=False, subok=True)
+    with warnings.catch_warnings():
+        warnings.filterwarnings('ignore', '', DeprecationWarning)
+        x = array(x, copy=False, subok=True)
+        y = array(y, copy=False, subok=True)
```

Comment:
hmm, is there a way around filtering the warning?

Reply:
Do we want to filter it? Can we instead fix the call sites?

Reply:
There are tests that do `np.assert_equal(a, [[1, 2], 3])`. Fixing the call sites would require them to do `np.assert_equal(a, np.array([[1, 2], 3], dtype=object)`. Would that be too much of a breaking change for people who use `np.assert_equal` and friends?

Reply:
IMO that's a reasonable thing to require them to do. As I understand it, the purpose of the NEP is to deprecate implicit ragged construction anywhere, not just in direct calls to `np.array`

Reply:
mentioning this in discussion of NEP 34

Reply:
reverted this and fixed tests where needed